pkg/cleaner/cleaning.go

```diff
@@ -0,0 +1,138 @@
+package cleaner
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/must-gather-clean/pkg/kube"
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"github.com/openshift/must-gather-clean/pkg/omitter"
+)
+
+type CleaningProcessor interface {
+	// ProcessFile is the end2end method for cleaning (omit + obfuscate path and content of) a file on disk.
+	// Will return nil if the file was processed without an error (e.g. through omission) or the error otherwise.
+	ProcessFile(inputFile string) error
+}
+
+type ReaderWriterObfuscator interface {
+	// ObfuscateInputReader obfuscates on an agnostic line-based reader and writes to an agnostic writer facility
+	ObfuscateInputReader(inputReader io.Reader, outputWriter io.Writer) error
+}
+
+type FileObfuscator interface {
+	// ObfuscatePlainTextFile obfuscates a text file and writes the result into the outputFile.
+	ObfuscatePlainTextFile(inputFile string, outputFile string) error
+}
+
+type ContentObfuscator struct {
+	Obfuscator obfuscator.ReportingObfuscator
+}
+
+type FileCleaner struct {
+	ContentObfuscator
```

Comment:
Is it really necessary to embed a struct which contains an interface?

Reply:
Also what's the point of using the `ReportingObfuscator` interface type here when it always has to be a `MultiObfuscator`.

Reply:
yeah that's exactly the place that I'm not happy about in the abstraction. I want to split the logic of obfuscating content and files. 

There's also no single place that requires a MultIObfuscator as the struct, the cleaner works on the ReportingObfuscator.

Reply:
> yeah that's exactly the place that I'm not happy about in the abstraction. I want to split the logic of obfuscating content and files.

it's probably clearer with the bash piping, since the ContentObfuscator is used there to obfuscate stdin. I don't want to duplicate the content obfuscation logic with the scanner once more in the fileCleaner. I'd prefer to have less duplication here - what's the problem with having a single value struct embedded?